ark/phenotyping/cell_cluster_utils.py

```diff
@@ -469,25 +462,29 @@ def cluster_cells(base_dir, cell_pysom, cell_som_cluster_cols,
             The SOM cluster object containing the cell SOM weights
         cell_som_cluster_cols (list):
             The list of columns used for SOM training
-        cell_som_cluster_count_avg_name (str):
-            The name of the file to write the clustered data
     """
 
     # raise error if weights haven't been assigned to cell_pysom
     if cell_pysom.weights is None:
         raise ValueError("Using untrained cell_pysom object, please invoke train_cell_som first")
 
+    # non-pixel cluster inputs won't be cell size normalized
+    cols_to_drop = ['fov', 'segmentation_label']
+    if 'cell_size' in cell_pysom.cell_data.columns.values:
+        cols_to_drop.append('cell_size')
+
     # ensure the weights columns are valid indexes, do so by ensuring
     # the cluster_counts_norm and weights columns are the same
     # minus the metadata columns that appear in cluster_counts_norm
     cluster_counts_size_norm = cell_pysom.cell_data.drop(
```

Comment:
Sorry, I think I'm being redundant with what Noah already said, but I only skimmed through his comments and thought I'd put this here anyways. Probably want to change the variable name of `cluster_counts_size_norm` to avoid the "cluster_counts" vocabulary.